girder/models/upload.py

```diff
@@ -172,10 +175,13 @@ def finalizeUpload(self, upload, assetstore=None):
         self.remove(upload)
 
         # Add an async event for handlers that wish to process this file.
-        events.daemon.trigger('data.process', {
+        eventParams = {
             'file': file,
             'assetstore': assetstore
-        })
+        }
+        if 'reference' in upload:
+            eventParams['reference'] = upload['reference']
+        events.daemon.trigger('data.process', eventParams)
```

Comment:
Arf, you're right. I didn't look to close and thought we were doing the same thing. Sorry for the noise.